module/ColorJudge.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file ColorJudge.cpp
+ * @brief 色識別クラス
+ * @author hiroto0927, miyashita64
+ */
+
+#include "ColorJudge.h"
+
+// RGBから色を判別する
+COLOR ColorJudge::getColor(rgb_raw_t const& _rgb)
+{
+  rgb_raw_t rgb;
+  hsv_raw_t hsv;
+  int min;
+
+  // 環境光による値の偏りを軽減する
+  // 事前に測った白が(255,255,255)となるように、RGB値を変換する
+  rgb.r = (_rgb.r < MAX_RGB.r) ? _rgb.r * 255 / MAX_RGB.r : 255;
+  rgb.g = (_rgb.g < MAX_RGB.g) ? _rgb.g * 255 / MAX_RGB.g : 255;
+  rgb.b = (_rgb.b < MAX_RGB.b) ? _rgb.b * 255 / MAX_RGB.b : 255;
```

Comment:
取得したRGB値を、0〜255の値に正規化したいということだよね？
去年・一昨年に使った正規化の式は、[このサイト](https://mathwords.net/dataseikika#_M_m)の式なんだよね。
でも、今書いてる式でもできてそうな感じはする。
僕じゃわからないから、@mutotaka0426 @Takahiro55555 @sugaken0528 @Hisataka-Hagiyama がどうにかして！
こういうときに、識別精度を数字で出していたら、判断がしやすいんだよね...

このサイトの式を当てはめるならば、
xmin = 事前に取得した黒のRGB = MIN_RGB
xmax = 事前に取得した白のRGB = MAX_RGB
X = 今回取得したRBG = _rgb
m = 0 
M = 255
として、
```suggestion
  rgb.r = (_rgb.r < MAX_RGB.r) ? (_rgb.r - MIN_RGB.r) * 255 / (MAX_RGB.r - MIN_RGB.r) : 255;
  rgb.g = (_rgb.g < MAX_RGB.g) ? (_rgb.g - MIN_RGB.g) * 255 / (MAX_RGB.g - MIN_RGB.g) : 255;
  rgb.b = (_rgb.b < MAX_RGB.b) ? (_rgb.b - MIN_RGB.b) * 255 / (MAX_RGB.b - MIN_RGB.b) : 255;
```

Reply:
@miyashita64 @hiroto0927 
2人がどういう資料を参考にこの式を書いているのか、またはどういう意図で書いているのか知りたいです。
xminを0にしたら @arimakaoru が添付してくださった資料の式になるみたいですけど0にしていいんですか？

Reply:
最低値を測って引く方が正しいと思います。
単純に最小値側の正規化を失念していました。